src/utils/components/SSHSecretSection/utils/utils.ts

```diff
@@ -41,24 +51,29 @@ export const detachVMSecret = async (vm: V1VirtualMachine) => {
   });
 };
 
-export const addSecretToVM = (vm: V1VirtualMachine, secretName?: string) =>
-  produce(vm, (vmDraft) => {
+export const addSecretToVM = (vm: V1VirtualMachine, secretName?: string, isDynamic?: boolean) => {
+  return produce(vm, (vmDraft) => {
     const cloudInitNoCloudVolume = getVolumes(vm)?.find((v) => v.cloudInitNoCloud);
+    const cloudInitData = getCloudInitData(getCloudInitVolume(vm));
+    const userData = convertYAMLUserDataObject(cloudInitData?.userData);
```

Comment:
I'd rather this would go inside the getCloudIntPropagationMethod, and called only if isDynamic is true